src/ngraph/runtime/cpu/cpu_external_function.cpp

```diff
@@ -1268,6 +1300,17 @@ void runtime::cpu::CPU_ExternalFunction::build(ngraph::pass::PassConfig& pass_co
     register_common_passes(pass_manager, pass_config);
     pass_manager.run_passes(m_function, false);
 
+    static const auto do_ngraph_trace = std::getenv("NGRAPH_TRACER_LOG");
```

Comment:
Add support to allow this to be set programmatically, not only via environment variables

Reply:
do you mean via some variable? can you provide me some basic idea to this?

Reply:
Here is an example of a set function https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/pass/manager.hpp#L72
That can programmatically set a variable that can get set via environment variable here
https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/pass/manager.cpp#L40-L44
This allows unit tests or other code to specifically set the flag without resorting to environment variables, but environment variables still work too.